lisa/features/gpu.py

```diff
@@ -126,23 +142,82 @@ def check_support(self) -> None:
         if not self._is_supported():
             raise SkippedException(f"GPU is not supported with distro {self._node.os}")
 
-    def install_compute_sdk(
-        self, driver: ComputeSDK = ComputeSDK.CUDA, version: str = ""
-    ) -> None:
+    def install_compute_sdk(self, version: str = "") -> None:
         # install GPU dependencies before installing driver
         self._install_gpu_dep()
 
+        if isinstance(self._node.os, Redhat):
+            # install LIS driver if not already installed.
+            self._node.tools[LisDriver]
+
         # install the driver
-        if driver == ComputeSDK.GRID:
-            if version == "":
-                version = DEFAULT_GRID_DRIVER_URL
-            self._install_grid_driver(version)
-        elif driver == ComputeSDK.CUDA:
-            if version == "":
-                version = "10.1.105-1"
-            self._install_cuda_driver(version)
-        else:
+        supported_driver = self._get_supported_driver()
+        for driver in supported_driver:
+            if driver == self.ComputeSDK.GRID:
+                if version == "":
+                    version = DEFAULT_GRID_DRIVER_URL
+                    self._install_grid_driver(version)
+                    self.gpu_vendor.add("nvidia")
+            elif driver == self.ComputeSDK.CUDA:
+                if version == "":
+                    version = "10.1.105-1"
+                    self._install_cuda_driver(version)
+                    self.gpu_vendor.add("nvidia")
+
+    def validate_gpu_count_with_lsvmbus(self, expected_count: int) -> None:
```

Comment:
It's better to provide count in feature, and validate in test case. So this method should return count from lsvmbus, and name is `get_gpu_count_by_lsvmbus`

Reply:
yeah I changed this so many times. get..() -> validate..() -> get..() -> validate..()
But now that I am seeing it, inside feature, get() makes more sense.